firebase-common/src/main/java/com/google/firebase/heartbeatinfo/HeartBeatInfo.java

```diff
@@ -44,4 +49,10 @@ public int getCode() {
 
   @NonNull
   HeartBeat getHeartBeatCode(@NonNull String heartBeatTag);
+
+  @NonNull
+  Task storeHeartBeatInfo(@NonNull String heartBeatTag);
```

Comment:
why is it a raw `Task`? should it be `Task<Boolean>`? also consider renaming the method to something like `shouldStoreHeardBeatInfo()`

Reply:
hmm the method itself would storeHeartBeatInformation. It returns a task so that we can wait on the task to complete if needed. The return type doesn't matter in this case but TaskCompletionSource requires a returnType.

Reply:
in this case use `Task<Void>`, but `raw types` break type safety